File: PythonSolutions/MaxProfit_121.py
from typing import List

class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        left = 0
        right = 1
        maxProfit = 0

        while right < len(prices):
            if prices[left] < prices[right]:
                profit = prices[right] - prices[left]
                maxProfit = max(maxProfit, profit)
            else:
                left = right
            right += 1

        return maxProfit
    

if __name__ == "__main__":
    solution = Solution()
    prices = [7, 1, 5, 3, 6, 4]
    print(solution.maxProfit(prices))
    

File: JavaScriptSolutions/MaxProfit_121.js
const maxProfit = (prices) => {
  let left = 0;
  let right = 1;
  let profit = 0;

  while (right < prices.length) {
    if (prices[left] < prices[right]) {
      let currentProfit = prices[right] - prices[left];
      profit = Math.max(profit, currentProfit);
    } else {
      left = right;
    }
    right += 1;
  }

  return profit;
};

const prices = [7, 1, 5, 3, 6, 4];

console.log(maxProfit(prices));
